proposals/unified-show-diff.md

```diff
@@ -0,0 +1,134 @@
+# Title: Unified show/diff command for all output types
+
+- Enhancement Proposal PR: (leave this empty)
+- Contributors: dberenbaum
```

Comment:
TBH I didn't have time to all the existing discussion so I hope the latest text I read reflects the main outcomes of those! Plus perhaps 👇 

```suggestion
- Enhancement Proposal PR: https://github.com/iterative/enhancement-proposals/pull/4
- Contributors: dberenbaum, pmrowla
```